modules/standard/IO.chpl

```diff
@@ -8390,7 +8393,7 @@ proc fileReader.readBinary(ref data: [?d] ?t, endian: ioendian):bool throws
 */
 proc fileReader.readBinary(ref data: [?d] ?t, endian: ioendian):int throws
   where ReadBinaryArrayReturnInt == true &&
-    d.rank == 1 && d.isRectangular() && d.hasUnitStride() && (
+    d.rank == 1 && d.isRectangular() && d.strides == strideKind.one && (
```

Comment:
Some of these where clauses refer to the array (e.g., `data.rank == 1`) and others, like this one, to the domain.  I'd suggest unifying on the array (and potentially also removing the domain queries when they're not used by the body of the routine?)